pkg/netns/netns.go

```diff
@@ -4,88 +4,302 @@
 package netns
 
 import (
-	"errors"
 	"fmt"
 	"os"
-	"os/exec"
+	"path"
 	"path/filepath"
+	"runtime"
 	"strings"
+	"sync"
 
-	"github.com/containernetworking/plugins/pkg/ns"
-	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+
+	// TODO consider dropping this dep entirely - it is only used here now
+	"github.com/vishvananda/netns"
 )
 
-// RemoveIfFromNetNSIfExists removes the given interface from the given
-// network namespace.
+const bindMountPath = "/run/netns" /* Bind mount path for named netns */
+
+type netNSHandle struct {
+	handle int
+	Name   string
+}
+
+type NetNsHandle interface {
+	// Returns a file descriptor representing this object's network namespace
+	Fd() int
+
+	// Cleans up this instance of the network namespace; if this instance
```

Comment:
Yep, as with new they should be separate. 

To your point tho, I do not know yet if we actually *need* to pin across process anywhere - except for health endpoint, where we do retain the FD in the creating process and do explicitly clean it up.

If there are explicitly no places in the code where a pinned netns needs to outlive the creating process, it may be cleaner to do this, even if it makes this less useful as a generic netns library.

Reply:
> where we do retain the FD in the creating process until after the secondary process terminates, and do explicitly clean it up

We need to clean up because `ip netns add` creates file system state. (`ip` is a separate process in and of itself, so it kinda has to)  If we'd simply launch the process in an unpinned netns, we wouldn't need to clean anything up to begin with. :slightly_smiling_face: And if the Pod dies, so does the health listener.

There may be folks still running Cilium outside of containers today, so as long as we keep the pidfile management code in the health endpoint launcher, we can always kill the old process before starting a new one, and the netns will die along with it.

Reply:
Yep, I was going to actually try launching the subproc unpinned, if that works agree we may not need pins at all (assuming I don't find other weird usages in the rest of the code)